src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2315,6 +2315,39 @@ theorem msb_shiftLeft {x : BitVec w} {n : Nat} :
     (x <<< n).msb = x.getMsbD n := by
   simp [BitVec.msb]
 
+/--
+A `(x : BitVec v)` set to width `w` equals `(v - w)` zeros,
+followed by the low `(min v w) bits of `x`
+-/
+theorem setWidth_eq_append_extractLsb' {v : Nat} {x : BitVec v} {w : Nat} :
+    x.setWidth w = ((0#(w - v)) ++ x.extractLsb' 0 (min v w)).cast (by omega) := by
+  ext i hi
+  simp only [getElem_setWidth, getElem_cast, getElem_append]
+  by_cases hiv : i < v
+  · simp [hi]
+    omega
+  · simp only [getElem_extractLsb', Nat.zero_add, getElem_zero, dite_eq_ite, Bool.if_false_right,
+    Bool.iff_and_self, decide_eq_true_eq]
+    intros hcontra
+    have : x.getLsbD i = false := getLsbD_ge x i (by omega)
+    simp [this] at hcontra
+
+/--
+A `(x : BitVec v)` set to a width `w ≥ v` equals `(w - v)` zeros, followed by `x`.
+-/
+theorem setWidth_eq_concat {v : Nat} {x : BitVec v} {w : Nat} (h : v ≤ w) :
+    x.setWidth w = ((0#(w - v)) ++ x).cast (by omega) := by
```

Comment:
I am a bit confused, these both use `++` on the RHS, but one is named `append` and the other `concat`.

Reply:
Thanks, pushed a fix!